src/main/java/de/chojo/jdautil/pagination/bag/PageBag.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *     SPDX-License-Identifier: AGPL-3.0-only
+ *
+ *     Copyright (C) 2022 RainbowDashLabs and Contributor
+ */
+
+package de.chojo.jdautil.pagination.bag;
+
+import de.chojo.jdautil.pagination.exceptions.EmptyPageBagException;
+
+/**
+ * A default page bag which circles through the pages.
+ */
+public abstract class PageBag implements IPageBag {
+    private final int pages;
+    private int currentPage = 0;
+
+    public PageBag(int pages) {
+        if (pages == 0) throw new EmptyPageBagException();
+        this.pages = pages;
+    }
+
+    @Override
+    public int pages() {
+        return pages;
+    }
+
+    @Override
+    public int current() {
+        return currentPage;
+    }
+
+    @Override
+    public void scrollNext() {
+        if (currentPage + 1 == pages) {
+            currentPage = 0;
+        } else {
+            currentPage++;
+        }
+    }
+
+    @Override
+    public boolean hasNext() {
+        return true;
+    }
```

Comment:
In my opinion, the method should actually check if a next page is available

Reply:
The Page Bag is implemented as a rolling bag. It will start at zwei if it exceeds the page size. Thats why this value is actually a constant.

Reply:
👍 Then you can resolve these two threads (I don't have permission to do so here)